pkg/platform/kube/functionres/lazy.go

```diff
@@ -570,6 +607,106 @@ func (lc *lazyClient) waitFunctionDeploymentReadiness(ctx context.Context,
 	return errors.New("Function deployment is not ready yet"), ""
 }
 
+// getFunctionDeployment returns function's deployment
+func (lc *lazyClient) getFunctionDeployment(ctx context.Context, function *nuclioio.NuclioFunction) (*appsv1.Deployment, error) {
```

Comment:
This wrapper function might be a bit redundant?

Reply:
@TomerShor why? It can be reusable

Reply:
I've added usage of it in ` waitFunctionDeploymentReadiness` functions